.github/workflows/test.yml

```diff
@@ -0,0 +1,37 @@
+name: Test
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    container:
+      image: golang:1.17.6
+
+    services:
+      mysql:
+        image: mysql:5.7
+        env:
+          MYSQL_ROOT_PASSWORD: mattermod
+          MYSQL_DATABASE: mattermod
+          MYSQL_USER: mattermod
+          MYSQL_PASSWORD: mattermod
+        ports:
+        - 3306:3306
+        # needed because the mysql container does not provide a healthcheck
+        options: --health-cmd="mysqladmin ping" --health-interval=10s --health-timeout=10s --health-retries=10
```

Comment:
Does this ensure the "wait till MySQL is ready"?

Reply:
during all the running i did to test it was always up, but we can add a check just to make sure

Reply:
actually, we don't need to do anything gh does all the checks

![Screenshot 2022-01-21 at 09 08 36](https://user-images.githubusercontent.com/4115580/150490175-68361b01-fcaa-47c2-9694-bc6860e2e48b.png)